packages/core/src/sessionManagement.ts

```diff
@@ -89,25 +123,46 @@ function retrieveSession(sessionCookie: CookieCache): SessionState {
   return session
 }
 
-export function persistSession(session: SessionState, cookie: CookieCache) {
+export function persistSession(session: SessionState, cookie: CookieCache, withNewSessionStrategy = false) {
+  if (withNewSessionStrategy && !utils.isEmptyObject(session)) {
+    session.expire = String(Date.now() + SESSION_EXPIRATION_DELAY)
+  }
   const cookieString = utils
     .objectEntries(session)
     .map(([key, value]) => `${key}=${value}`)
     .join(SESSION_ENTRY_SEPARATOR)
-  cookie.set(cookieString, EXPIRATION_DELAY)
+  cookie.set(cookieString, SESSION_EXPIRATION_DELAY)
 }
 
 export function stopSessionManagement() {
-  registeredActivityListeners.forEach((e) => e())
-  registeredActivityListeners = []
+  registeredListeners.forEach((e) => e())
+  registeredIntervals.forEach((interval) => clearInterval(interval))
+  registeredListeners = []
+  registeredIntervals = []
 }
 
-let registeredActivityListeners: Array<() => void> = []
+let registeredListeners: Array<() => void> = []
+let registeredIntervals: number[] = []
 
 export function trackActivity(expandOrRenewSession: () => void) {
+  const doExpandOrRenewSession = monitor(expandOrRenewSession)
   const options = { capture: true, passive: true }
   ;['click', 'touchstart', 'keydown', 'scroll'].forEach((event: string) => {
-    document.addEventListener(event, expandOrRenewSession, options)
-    registeredActivityListeners.push(() => document.removeEventListener(event, expandOrRenewSession, options))
+    document.addEventListener(event, doExpandOrRenewSession, options)
+    registeredListeners.push(() => document.removeEventListener(event, doExpandOrRenewSession, options))
+  })
+}
+
+function trackVisibility(expandSession: () => void, visibilityStateProvider: () => VisibilityState) {
+  const expandSessionWhenVisible = monitor(() => {
+    if (visibilityStateProvider() === 'visible') {
+      expandSession()
+    }
   })
+
+  const visibilityCheckInterval = window.setInterval(expandSessionWhenVisible, VISIBILITY_CHECK_DELAY)
+  document.addEventListener('visibilitychange', expandSessionWhenVisible)
+
+  registeredIntervals.push(visibilityCheckInterval)
+  registeredListeners.push(() => document.removeEventListener('visibilitychange', expandSessionWhenVisible))
```

Comment:
Could `'visibilitychange'` be a constant?

Reply:
sure, what is your reasoning?
- avoid to duplicate this value inside this function
- have constant for all events we are listening

Reply:
I'd say both 😄

Reply:
But in the scope of that PR only inside that function.